src/hooks/useAuth.ts

```diff
@@ -0,0 +1,55 @@
+import { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import * as authApi from '@/apis/auth';
+import { LoginRequestBody } from '@/apis/types';
+import { useUserState } from '@/atoms/userState';
+import { loginError } from '@/constants/error';
+import Uri from '@/constants/uri';
+
+const useAuth = (): {
+  login: ({ id, password }: LoginRequestBody) => Promise<void>;
+  logout: () => void;
+  isLoading: boolean;
+  error: string;
+} => {
+  const [, setUserState] = useUserState();
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
```

Comment:
훅을 사용하는 코드에서 다뤄야 할 에러가 많아질 수 있으니 authError라는 네이밍을 사용하는건 어떨까요?

```suggestion
  const [authError, setAuthError] = useState<string | null>(null);
```

Reply:
그런 문제는 해당 hook에서 관리해야 할 문제가 아니라고 생각합니다. 그런 관점에서 접근을 한다면 loading 관련된 문제도 `auth`라는 prefix를 붙여야 할테고 return 하는 모든 값들에 대해 prefix가 필요할 것입니다.  이러한 처리들은 불필요하고 네이밍적인 복잡함만 초래할 수 있기 때문에 중복되는 값들에 대해서는 아래와 같이 처리하는 것이 좋다고 생각합니다.

```ts
const { error: authError } = useAuth();
```

Reply:
오... 납득되네요 좋습니다 👍🏻